### ISSUE TITLE
Feature: embed image/logo

### ISSUE THREAD

**Issue by tovine** [nayuki/QR-Code-generator#27]
Many other QR generators (typically online versions) give an option to trade some redundancy data for the option of embedding an image in the middle of the QR code - would this be possible?

Just an idea for a "nice to have" feature, probably shouldn't be too hard to implement (apart from validating the image to make sure it doesn't ruin the readability of the data in any way).

The drawing itself could be done similarly to what's discussed in this answer: https://stackoverflow.com/a/11148571 seeing as you already use `BufferedImage` in the `toImage` function.

**Comment by nayuki** [comment]
Thanks for the suggestion; I am aware that this kind of image branding is common. I would argue that this isn't kosher with respect to the QR Code standard, and opens the box to all sorts of options and customizations. I would decline this feature.

**Comment by tovine** [comment]
I would argue that you can know exactly how many pixels were blocked by specifying the size of the overlay yourself. 
However on a second thought I agree that it doesn't necessarily belong in this library, and can just as easily be implemented in a thin wrapper.

**Comment by Bitlitas** [comment]
Yeah, we need logo in center :+1:

**Comment by rugk** [comment]
BTW theoretically this is totally possible in an easy way. In most cases you can just place it in the middle. The error correction itself correctly handles the pixels, which are overlapped by the image.

See https://en.wikipedia.org/wiki/QR_code#Error_correction (have a look at the images)
